corehq/apps/userreports/tasks.py

```diff
@@ -178,25 +186,52 @@ def _run_report(backend_to_use):
     return objects
 
 
+@periodic_task(
+    run_every=ASYNC_INDICATOR_QUEUE_TIME,
+    queue=settings.CELERY_PERIODIC_QUEUE,
+)
+def queue_async_indicators():
+    start = datetime.utcnow()
+    cutoff = start + ASYNC_INDICATOR_QUEUE_TIME
+    time_for_crit_section = ASYNC_INDICATOR_QUEUE_TIME.seconds - 10
+
+    with CriticalSection(['queue-async-indicators'], timeout=time_for_crit_section):
+        redis_client = get_redis_client().client.get_client()
+        indicators = AsyncIndicator.objects.all()[:10000]
+        paginator = Paginator(indicators, 1000)
+        for page in paginator.page_range:
+            for indicator in paginator.page(page).object_list:
+                now = datetime.utcnow()
+                if now > cutoff:
```

Comment:
never loved the way repeaters used this to cut off the task. maybe this is slightly better and more reliable? http://docs.celeryproject.org/en/latest/userguide/configuration.html#std:setting-task_soft_time_limit